### PR TITLE
Migrate from Travis to GitHub Actions

### LINKED ISSUE #67: Migrate from Travis CI to GitHub Actions
Replace the existing Travis CI configuration with one based on GitHub Actions. Ensure that:

- Matrix builds are preserved
- Caches are setup appropriately
- Tagged releases are automatically published to pypi

### PR DESCRIPTION
**Pull request by rbreslow**
## Overview

Ports Travis CI config to a GitHub Actions workflow.

Closes #67  

### Notes

The `PYPI_AZAVEA_USERNAME` and `PYPI_AZAVEA_PASSWORD` secrets have been made available to this repository.

## Testing Instructions

 * Verify that the [GitHub Actions CI workflow](https://github.com/azavea/django-amazon-ses/actions/runs/206073800) passes.
* Verify that [this Release workflow](https://github.com/azavea/django-amazon-ses/runs/977540454?check_suite_focus=true) ran. The workflow was expected to fail at the "Build and publish package" step since the PyPI credentials had not been shared with this repository yet.

### CODE REVIEW

File: .github/workflows/continuous_integration.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - master
+  pull_request:
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.6", "3.7", "3.8"]
```
Comment (coalkettler) [inline .github/workflows/continuous_integration.yml]:
```suggestion
        python-version: ["3.5", "3.6", "3.7", "3.8"]
```

We'll need to include this version for `lint` to run in `tox`.

3.5 is pretty close to EOL and I'd be fine with dropping (set up #71 to track it), but I recognize that this is module has a larger user base than some of our other projects. What's your thoughts?

**Review by coalkettler — COMMENTED** [review]
Left some feedback for ya!

File: .github/workflows/continuous_integration.yml (hunk 1 above)
Comment (rbreslow) [reply .github/workflows/continuous_integration.yml]: I think it makes sense to keep 3.5 support.

**Review by coalkettler — APPROVED** [review]
LGTM! ⛵

**Pull request merged**